components/icons/icon.js

```diff
@@ -0,0 +1,145 @@
+import { css, html, LitElement } from 'lit-element/lit-element.js';
+import { loadSvg } from '../../.generated/icons/presetIconLoader.js';
+import { RtlMixin } from '../localize/rtl-mixin.js';
+import { unsafeHTML } from 'lit-html/directives/unsafe-html.js';
+
+export class D2LIcon extends RtlMixin(LitElement) {
+
+	static get properties() {
+		return {
+			icon: {
+				type: String,
+				reflect: true
+			},
+			size: {
+				type: String,
+				reflect: true
+			},
+			src: {
+				type: String,
+				reflect: true
+			},
+			__imgSrc: { type: String },
+			__svg: { type: Object }
+		};
+	}
+
+	constructor() {
+		super();
+		this.size = 'tier1';
+	}
+
+	static get styles() {
+		return css`
+			:host {
+				-ms-flex-align: center;
+				-webkit-align-items: center;
+				align-items: center;
+				color: var(--d2l-color-ferrite);
+				display: -ms-inline-flexbox;
+				display: -webkit-inline-flex;
+				display: inline-flex;
+				fill: var(--d2l-icon-fill-color, currentcolor);
+				height: var(--d2l-icon-height, 18px);
+				-ms-flex-pack: center;
+				-webkit-justify-content: center;
+				justify-content: center;
+				stroke: var(--d2l-icon-stroke-color, none);
+				vertical-align: middle;
+				width: var(--d2l-icon-width, 18px);
+			}
+			:host([hidden]) {
+				display: none;
+			}
+			:host([icon*="d2l-tier2:"]),
+			:host([size="tier2"]) {
+				height: var(--d2l-icon-height, 24px);
+				width: var(--d2l-icon-width, 24px);
+			}
+			:host([icon*="d2l-tier3:"]),
+			:host([size="tier3"]) {
+				height: var(--d2l-icon-height, 30px);
+				width: var(--d2l-icon-width, 30px);
+			}
+			svg, img {
+				display: block;
+				height: 100%;
+				pointer-events: none;
+				width: 100%;
+			}
+			:host([dir="rtl"]) svg[mirror-in-rtl] {
+				-webkit-transform: scale(-1,1);
+				transform: scale(-1,1);
+				transform-origin: center;
+			}
+		`;
+	}
+
+	attributeChangedCallback(name, oldval, newval) {
+		if (name === 'icon') {
+			this.__fetchSvg(newval);
+		} else if (name === 'src') {
+			if (newval && newval.substr(newval.length - 4) === '.svg') {
+				this.__fetchSrcSvg(newval);
+			} else {
+				this.__svg = undefined;
+				this.__imgSrc = newval;
+			}
+		}
+		super.attributeChangedCallback(name, oldval, newval);
+	}
+
+	shouldUpdate(changedProperties) {
+		const shouldUpdate = changedProperties.has('__svg')
+			|| changedProperties.has('__imgSrc');
```

Comment:
We only care if one of these two properties change, since they're all that should cause a re-render.